irc_nicktrack.go

```diff
@@ -0,0 +1,103 @@
+package irc
+
+import (
+	"regexp"
+	"strings"
+)
+
+//Struct to store Channel Info
+type Channel struct {
+	Topic string
+	Mode  string
+	Users map[string]User
+}
+
+type User struct {
+	Host string
+	Mode string
+}
+
+var mode_split = regexp.MustCompile("([%@+]{0,1})(.+)") //Half-Op, //Op, //Voice
```

Comment:
This is fairly limited and will break on any servers that support other modes (I've seen & used for admin). This should be pulled from the `PREFIX` isupport value which comes in the format like this: `(qaohv)~&@%+`